lfucg-exactions/client/js/components/AccountLedgerForm.js

```diff
@@ -251,11 +316,126 @@ function mapDispatchToProps(dispatch, params) {
+                    });
+                }
+
+                if (field === 'plat_lot') {
+                    if (value_id === 'plat') {
+                        dispatch(getPlats());
+                    } else if (value_id === 'lot') {
+                        dispatch(getLots());
+                    }
+                }
+            };
+        },
+        platFormChange(field) {
+            return (e, ...args) => {
+                const value = typeof e.target.value !== 'undefined' ? e.target.value : args[1];
+
+                const comma_index = value.indexOf(',');
+                const dollar_index = value.indexOf('$');
+                const second_dollar_index = value.indexOf(',$', dollar_index + 1);
+
+                const value_id = value.substring(0, comma_index);
+                const value_name = value.substring(comma_index + 1, dollar_index);
+                const value_non_sewer = value.substring(dollar_index, second_dollar_index);
+                const value_sewer = value.substring(second_dollar_index + 1, value.length);
+
+                const field_name = `${[field]}_name`;
+                const field_show = `${[field]}_show`;
+
+                const plat_update = {
+                    [field]: value_id,
+                    [field_name]: value_name,
+                    [field_show]: value,
+                    non_sewer_exactions: value_non_sewer,
+                    sewer_exactions: value_sewer,
+                };
+
+                dispatch(formUpdate(plat_update));
+            };
+        },
+        lotFormChange(field) {
+            return (e, ...args) => {
+                const value = typeof e.target.value !== 'undefined' ? e.target.value : args[1];
+
+                const comma_index = value.indexOf(',');
+                const dollar_index = value.indexOf('$');
+                const second_dollar_index = value.indexOf(',$', dollar_index + 1);
+
+                const value_id = value.substring(0, comma_index);
+                const value_name = value.substring(comma_index + 1, dollar_index);
+                const value_non_sewer = value.substring(dollar_index, second_dollar_index);
+                const value_sewer = value.substring(second_dollar_index + 1, value.length);
+
+                const field_name = `${[field]}_name`;
+                const field_show = `${[field]}_show`;
+
+                const lot_update = {
+                    [field]: value_id,
+                    [field_name]: value_name,
+                    [field_show]: value,
+                    non_sewer_exactions: value_non_sewer,
+                    sewer_exactions: value_sewer,
+                };
+
+                dispatch(formUpdate(lot_update));
+            };
+        },
+        accountFormChange(field) {
+            return (e, ...args) => {
+                const value = typeof e.target.value !== 'undefined' ? e.target.value : args[1];
+
+                const comma_index = value.indexOf(',');
```

Comment:
Need to make sure you actually got a valid index before using these index variables